test/test.js

```diff
@@ -211,12 +284,16 @@ describe('vhost(hostname, server)', function () {
   })
 })
 
-function createServer (hostname, server) {
+function createServer (hostname, server, pretest) {
   var vhosts = !Array.isArray(hostname)
     ? [vhost(hostname, server)]
     : hostname
 
-  return http.createServer(function onRequest (req, res) {
+  // This allows you to perform changes to the request/response
+  // objects before our assertions
+  pretest = pretest || function () {}
```

Comment:
Typically, you don't want to reassign variables that represent function arguments.